### DISCUSSION TITLE
Cloud/Virtual Files

### DISCUSSION THREAD

**Discussion by bionemesis** [borgbase/vorta#1098] [General]
Has there been any discussion around Vorta's handling of virtual files like OneDrive's on-demand feature? I was just testing out Vorta and realized that as it hits a OneDrive folder, it will trigger OneDrive to download files locally. That's not ideal. We can work around the issue with exclusion filters, but that's really not a full solution. I don't believe other backup products experience this issue.

**Comment by m3nu** [comment]
I don't know about OneDrive, but Google Drive with the "streaming" option enabled would mount as new file system. So using the "one filesystem" option would exclude such files.

Is there any file attribute or other way Borg could use to find such files? Else this is more an issue for Borg, since files are being read there.

**Reply by bionemesis** [reply]
Good point. I've posted the question to the Borg github discussions. I'm not sure if OneDrive does a separate filesystem like Google Drive. I suspect not.

**Reply by bionemesis** [reply]
So it turns out the only way to identify these files is via xattrs and the com.apple.metadata:com_apple_backup_excludeItem attribute. Unfortunately, it appears that borg has no plans to support including/excluding based on xattrs according to @ThomasWaldmann.

**Reply by ThomasWaldmann** [reply]
Well, there are basically 2 options:

- the general solution would be to extend include/exclude filename-based pattern definitions to be way more general so they can also match based on other file metadata, like xattrs or size or ... - planning and implementing this will need quite some time and work.
- making a quite special feature for macOS users that checks precisely the mentioned xattr - that would be way less effort.
